modules/gcp/compute.go

```diff
@@ -581,10 +584,23 @@ func NewComputeService(t *testing.T) *compute.Service {
 func NewComputeServiceE(t *testing.T) (*compute.Service, error) {
 	ctx := context.Background()
 
-	client, err := google.DefaultClient(ctx, compute.CloudPlatformScope)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to get default client: %v", err)
-	}
+	// Retrieve the Google OAuth token using a retry loop as it can sometimes return an error.
+	// e.g: oauth2: cannot fetch token: Post https://oauth2.googleapis.com/token: net/http: TLS handshake timeout
+	// This is loosely based on https://github.com/kubernetes/kubernetes/blob/7e8de5422cb5ad76dd0c147cf4336220d282e34b/pkg/cloudprovider/providers/gce/gce.go#L831.
+	maxRetries := 6
+	timeBetweenRetries := 10 * time.Second
+	var client *http.Client
+	var err error
```

Comment:
Can you add a log message with the retry params, so we log how long this might try?

Reply:
Sure